methods/array_ops/src/pg_gp/array_ops.c

```diff
@@ -1929,3 +1971,126 @@ General_Array_to_Array(
     return pgarray;
 }
 
+/******* internal functions ********/
```

Comment:
Thanks for the pointer. I agree that the current implementation is a bit too complicated since it relies on the low-level c layer.